azurelinuxagent/ga/update.py

```diff
@@ -1017,6 +1023,49 @@ def _ensure_firewall_rules_persisted(dst_ip):
             message=msg,
             log_event=False)
 
+    def _add_accept_tcp_firewall_rule_if_not_enabled(self, dst_ip):
+        # Helper to execute a run command, returns True if no exception
+        # Here we primarily check if an  iptable rule exist. True if it exits , false if not
+        def _execute_run_command(command):
+            try:
+                shellutil.run_command(command)
+                return True
+            except CommandError as e:
+                # return code 1 is expected while using the check command. Raise if encounter any other return code
+                if e.returncode != 1:
+                    raise
+            return False
+
+        try:
+            wait = self.osutil.get_firewall_will_wait()
+
+            # "-C" checks if the iptable rule is available in the chain. It throws an exception with return code 1 if the ip table rule doesnt exist
+            drop_rule = get_firewall_drop_command(wait, AddFirewallRules.CHECK_COMMAND, dst_ip)
+            if not _execute_run_command(drop_rule):
+                # DROP command doesn't exist indicates then none of the firewall rules are set yet
+                # exiting here as the environment thread will set up all firewall rules
+                return
+            else:
+                # DROP rule exists in the ip table chain. Hence checking if the DNS TCP to wireserver rule exists. If not we add it.
+                accept_tcp_rule = get_accept_tcp_rule(wait, AddFirewallRules.CHECK_COMMAND, dst_ip)
+                if not _execute_run_command(accept_tcp_rule):
+                    try:
+                        logger.info(
+                            "Firewall rule to allow DNS TCP request to wireserver for a non root user unavailable . Setting it now.")
+                        accept_tcp_rule = get_accept_tcp_rule(wait, AddFirewallRules.INSERT_COMMAND, dst_ip)
+                        shellutil.run_command(accept_tcp_rule)
+                        logger.info(
+                            "Succesfully added firewall rule to allow non root users to do a DNS TCP request to wireserver ")
+                    except Exception as e:
```

Comment:
NIT: This should be a check for `CommandError` and not `Exception` as the only thing that can potentially throw here is `shellutil.run_command`

Reply:
Addressed